File: 0242. Valid Anagram/0242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }

        std::unordered_map<char, int> myMap;

        for (char c : s) {
            myMap[c]++;
        }
        for (char c : t) {
            myMap[c]--;
            if (myMap[c] < 0) {
                return false;
            }
        }
        return true;
    }
};

File: Arrays & Hashing/0242. Valid Anagram/0242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        hashmap = defaultdict(int)
        
        for i in s:
            hashmap[i] += 1

        for i in t:
            hashmap[i] -= 1
            if hashmap[i] < 0:
                return False
        
        return True
        
        
